jina/executors/indexers/cache.py

```diff
@@ -62,7 +62,7 @@ def __init__(self, index_filename: str = None, *args, **kwargs):
             raise ValueError(f"Field '{self.field}' not in supported list of {self.supported_fields}")
 
     @deprecated_alias(doc_id='doc')
-    def add(self, doc: 'Document', *args, **kwargs):
+    def add(self, doc, *args, **kwargs):
```

Comment:
u need to have the trick 

if False:
from jina import Document

Reply:
That fails 

> > * Make DocIDCache capable of detecting collisions on content-level.
> > * NOTE: Tests are only on the `Indexer` level, since the Drivers for Update/Del. haven't been implemented yet. Integration tests will be required when that is done.
> > 
> > **Discussion required:** do we set the `field` (`id`/`content_hash`) at the Driver level or at the Indexer level? Atm it's done at the Driver level.
> > NOTE: This is a replacement of #1466 because that one was failing the commitlint
> 
> I think is better to have it at Driver level. It is better to keep the Executor unaware of any Jina data type.

That was the original approach. Two problems:

1. Since it doesn't know which field it's caching, we can't optimize memory/space usage to avoid duplicating the ids (when we are caching by id)
2. It requires that whenever a user wants to use `DocIDCache` with `content_hash` (no the default option, `id`) to have to replicate the yml to specify it at the Driver level as parameter:


```yaml
!DocIDCache
with:
  index_path: cache.tmp
metas:
  name: cache
requests:
  on:
    [SearchRequest, TrainRequest, IndexRequest, ControlRequest]:
      - !RouteDriver {}
    IndexRequest:
      - !TaggingCacheDriver
        with:
          field: content_hash
          tags:
            is_indexed: true
      - !FilterQL
        with:
          lookups: {tags__is_indexed__neq: true}
```

vs

```yaml
!DocIDCache
with:
  index_path: cache.tmp
  field: content_hash
metas:
  name: cache
```

Reply:
That failed before. I rebased on master again, and it seems to have been fixed. Yep, will do